pkg/container/qemu_runner.go

```diff
@@ -126,7 +126,7 @@ func (bw *qemu) Debug(ctx context.Context, cfg *Config, envOverride map[string]s
 	if pubKey != nil {
 		command := fmt.Sprintf("echo %q | tee -a /root/.ssh/authorized_keys /home/*/.ssh/authorized_keys", strings.TrimSpace(string(pubKey)))
 		err := sendSSHCommand(ctx,
-			cfg.WorkspaceClient,
+			cfg.SSHControlClient,
```

Comment:
Shouldn't we use the build client here? I would expect in this connection to mimic what happens at build/test time. For instance, when specifying a non-root run-as user, I would expect to debug with the same user. That would apply also to the chroot-ed environment, I think.